### PR TITLE
Slicing 

### PR DESCRIPTION
**Pull request by partvishegy**
- add logging
- add slicing and segmentation
- missing doctests and unittest - sry, they are coming
- currently checks if transcripts exist for new input files, but does not care about interim states

### CODE REVIEW

File: src/transcript_tool.py [hunk 1]
```diff
@@ -7,9 +7,41 @@
...
+
+# show logs on the console as well (for now). this may get behind a -v flag
+root = logging.getLogger()
+handler = logging.StreamHandler(sys.stdout)
+handler.setLevel(logging.DEBUG)
+formatter = logging.Formatter("%(asctime)s - %(name)s - %(levelname)s - %(message)s")
+handler.setFormatter(formatter)
+root.addHandler(handler)
 
 DEFAULT_MODEL = "../models/default_model/"
```
Comment (zolizoli) [inline src/transcript_tool.py]: TODO: szedje majd le a modellt valahonnét (lehetőleg a huggingface hubról)

File: src/transcript_tool.py [hunk 2]
```diff
@@ -56,39 +88,153 @@ def main(in_path: str, out_path: str, model_path: str) -> None:
...
     # fetch model
     model = SpeechRecognitionModel(model_path)
-    print(f"\nModel: {model_path} fetched\n")
+    logging.info("Model loaded from: %s", model_path)
+
+    # create interim folders for processing slices & segments
+    if not os.path.isdir("../data/interim/slices"):
+        os.makedirs(os.path.abspath("../data/interim/slices"))
+    if not os.path.isdir("../data/interim/segments"):
+        os.makedirs(os.path.abspath("../data/interim/segments"))
```
Comment (zolizoli) [inline src/transcript_tool.py]: TODO: add test; package install can run into troubles with permissions
Would be nicer to check permissions, too and explicitly raise exceptions (e.g. if permission is denied and etc).

File: src/transcript_tool.py [hunk 3]
```diff
@@ -56,39 +88,153 @@ def main(in_path: str, out_path: str, model_path: str) -> None:
...
+        if is_processed(mp3, out_path):
+            logging.info("%s already has a transcirpt in %s", mp3, out_path)
+            continue
         # create transcript
         make_transcript(mp3, out_path, model)
 
 
 def make_transcript(
     audio_file: str, out_path: str, model: SpeechRecognitionModel
-) -> None:  # what is the model type?
```
Comment (zolizoli) [inline src/transcript_tool.py:75]: use type inference to determine the type of any unknown type:
https://mypy.readthedocs.io/en/stable/type_inference_and_annotations.html#
reveal_type(X) will run type inference on X and prints out its type
NB: reveal_type() will only run during type checking (mypy) and should be removed when you wanna run the code

File: src/transcript_tool.py [hunk 4]
```diff
@@ -56,39 +88,153 @@ def main(in_path: str, out_path: str, model_path: str) -> None:
...
+
+
+def is_processed(mp3: str, out_path: str) -> bool:
+    "Check wether the given audio file already has a transcript."
+    _, base_name = separate_fileneme(mp3)
+    file_name = os.path.join(out_path, base_name + ".txt")
+    return os.path.isfile(file_name)
 
 
 if __name__ == "__main__":
```
Comment (zolizoli) [inline src/transcript_tool.py]: maybe we can refactor the functions into an other module and here we just import it and provide the CLI in main

**Review by zolizoli — COMMENTED** [review]
Good job. The functions have got clear responsibilities and seem to be testable.

File: src/transcript_tool.py (hunk 4 above)
Comment (partvishegy) [reply src/transcript_tool.py]: separated utility function to utils.py as a first step ([eba5a6d](https://github.com/Koffair/radioship_transcripter/pull/2/commits/eba5a6d6b8af4b60b5d15a69dc34dc268e1530d9))
(had some disagreement with mypy on the proper way to import, but figured out)

File: src/transcript_tool.py (hunk 1 above)
Comment (partvishegy) [reply src/transcript_tool.py]: már fent van?

File: src/transcript_tool.py (hunk 3 above)
Comment (partvishegy) [reply src/transcript_tool.py:75]: noted!

File: src/transcript_tool.py (hunk 2 above)
Comment (partvishegy) [reply src/transcript_tool.py]: i remember that we talked about this, but im not sure how i can create such a scenario. how do i force a permission error if the ../data folder is created by and within the package itself?

Comment (partvishegy) [reply src/transcript_tool.py]: moved interim folders under the output folder, also slicing and segmenting gets their output paths as an argument, so they are easier to test as well. currently writing the tests.

File: src/transcript_tool.py (hunk 1 above)
Comment (partvishegy) [reply src/transcript_tool.py]: done.

**Pull request merged**